### DISCUSSION TITLE
关于移动端表格

### DISCUSSION THREAD

**Discussion by awmleer** [ant-design/ant-design-mobile#4016] [Ideas]
移动端在某些情况下的确是需要表格的，antd-mobile 目前之所以没有做 Table 组件，是考虑到很难设计出一套很通用的表格组件，移动端的场景太复杂了，不像桌面端中后台的表格都是很固定的形式

大家如果有业务场景中用到了表格，可以在在下面的评论区贴一下～（如果不涉及敏感数据的话）

我们来一起讨论下如果要做 Table 组件的话，该如何做才比较好

**Comment by xiaohuoni** [comment]
我们现在项目中用到的表格及其api
<img src="https://user-images.githubusercontent.com/11746742/134448146-bc9f7a36-9640-4057-a7a5-846427b46a59.png" width="300px" />
### 参数

<API />

###
属性说明：

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| dataSource | 数据数组 | any[] | - |
| columns | 表格列的配置 | ColumnProps[] | - |
| twoDimension | 是否是二维表格 | boolean | false |

Column属性说明:

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| title | 标题 | React.ReactNode | - |
| dataIndex | 数据指向 | string | - |
| width | 列宽，百分比值 | number | - |
| key | 关键帧 | string | - |
| align | 对齐 | string，有'center/flex-start/flex-end' | 'flex-start' |
| render | 生成复杂数据的渲染函数，参数分别为当前行的值，当前行数据，行索引 | Function(text, record, index) {} | - |

**Comment by zhazhazhaoo** [comment]
![image](https://user-images.githubusercontent.com/30381630/134448231-5a899265-1a66-436f-bf2a-405551863636.png)
这个

**Reply by xiaohuoni** [reply]
这个直接用 antd 的 table 写过这样的。兼容和性能都还可以。除了不符合设计规范。其他都很好

**Reply by nnecec** [reply]
如果【表头放在左侧，纵列是 item 信息】这样呢？可以给个配置 选择表头在左侧还是顶部。

**Reply by zhazhazhaoo** [reply]
这个是UI图  不是项目开发的图

**Reply by xiaohuoni** [reply]
>【表头放在左侧，纵列是 item 信息】

还没见过左侧表头的，有遇到长表格，表头固定顶部的需求。

**Reply by chj-damon** [reply]
antd的Table底层是rc-table，它是基础的逻辑，跟UI没多大关系，所以可以基于rc-table，封装一个移动端的交互UI，来实现跟antd Table一样的效果。

**Reply by githubqh** [reply]
> 这个直接用 antd 的 table 写过这样的。兼容和性能都还可以。除了不符合设计规范。其他都很好

意思移动端是支持antd web端的table组件的是吗？

**Comment by xiaoliu1990** [comment]
公司有一个移动端的统计报表系统，需要h5看，有图表统计和表格统计
图表主要是负责统筹对比，看总体进度
表格主要是用来看单个人员的详细情况，一般横向在5个以上
需求有下面这些
1、分页（无限滚动的模式）
2、左右能够固定
3、横向滚动
4、排序
5、控制文字换行，超过省略号

**Comment by awmleer** [comment]
#3811

**Comment by awmleer** [comment]
#4096

**Comment by awmleer** [comment]
感觉可以直接用 [React Table](https://react-table.tanstack.com)？我们没必要再搞了

**Reply by dragonyun** [reply]
可是 React Table 是一个工具，并不是一个组件，使用成本高。还是希望能有一个表格组件方便使用，有没有什么好的移动端表格组件推荐也好

**Comment by ngwszsd** [comment]
https://visactor.io/vtable/guide/introduction 字节的vtable，我们正在向他迁移，刚排班下周做。
![image](https://github.com/user-attachments/assets/41f7e139-79ca-4298-80ca-7e460e775fa2) 这是我们之前手撸的大数据性能不行只能分页

**Comment by ykcrush** [comment]
能不能做出类似汽车之家里面的那个表格，
![汽车之家](https://github.com/user-attachments/assets/d2ddf079-2040-463e-b143-fe1c52b452c0)

**Comment by mustars** [comment]
都2026年了
